lib/customisation.nix

```diff
@@ -404,4 +405,96 @@ rec {
       };
     in self;
 
+  /*
+    Define a `mkDerivation`-like function based on another `mkDerivation`-like function.
+
+    `mkDerivation` gives access to its final set of derivation attributes when
+    it is passed a function, or when it is passed an overlay-style function in
+    `overrideAttrs`.
+
+    Instead of composing new `mkDerivation`-like build helpers using normal
+    function composition, `extendMkDerivation` makes sure that the returned
+    build helper supports such first class recursion like `mkDerivation` does.
+
+    the base build helper by changing the input arguments. The choice of the
```

Comment:
There is something missing from this sentence

Reply:
This one is no longer relevent and should be deleted. It's the remaining of https://github.com/NixOS/nixpkgs/pull/234651#issuecomment-1837269743.